manifests/config.pp

```diff
@@ -69,7 +73,42 @@
     purge   => $purge,
     recurse => $recurse,
   }
+ 
+  if $collectd_selinux and $::selinux == 'true' {
+    selboolean { 'collectd_tcp_network_connect':
+      persistent => true,
+      provider   => getsetsebool,
+      name       => collectd_tcp_network_connect,
+      value      => 'on',
+    }
+    file { 'collectd_udev.pp':
```

Comment:
A binary is not portable across distros and not even across distro minor versions. A module compiled on EL 7.3 doest necessarily work on EL7.1 and fore sure wont work on EL6 or on Ubuntu.